lib/secure_headers/middleware.rb

```diff
@@ -7,9 +7,8 @@ def initialize(app)
 
     # merges the hash of headers into the current header set.
     def call(env)
-      req = Rack::Request.new(env)
       status, headers, response = @app.call(env)
-
+      req = Rack::Request.new(env)
```

Comment:
I think this change will mean we're deferring the creation of the `req` object until it's actually needed.